animate/metric.py

```diff
@@ -87,6 +81,16 @@ def __init__(self, function_space, *args, **kwargs):
         entity_dofs[0] = tdim**2
         plex.setSection(mesh.create_section(entity_dofs))
 
+        self._variable_parameters = {
+            "dm_plex_metric_h_min": firedrake.Constant(1.0e-30),
+            "dm_plex_metric_h_max": firedrake.Constant(1.0e30),
+            "dm_plex_metric_a_max": firedrake.Constant(1.0e5),
+            "dm_plex_metric_boundary_tag": None,
+        }
+        self._variable_parameters_set = False
+        if self.metric_parameters:
+            self.set_parameters(self.metric_parameters)
```

Comment:
While we're here, I guess it would be worth adding a comment for this code block - something along the lines of "Process spatially variable metric parameters"?

Reply:
Fixed in 3add3e6.